modules/gdscript/gdscript_parser.cpp

```diff
@@ -1077,14 +1077,15 @@ GDScriptParser::Node *GDScriptParser::_parse_expression(Node *p_parent, bool p_s
 						}
 						expecting = DICT_EXPECT_COMMA;
 
-						if (key->type == GDScriptParser::Node::TYPE_CONSTANT) {
-							Variant const &keyName = static_cast<const GDScriptParser::ConstantNode *>(key)->value;
-
-							if (keys.has(keyName)) {
-								_set_error("Duplicate key found in Dictionary literal");
+						const Variant *key_value = _try_to_find_constant_value_for_expression(key);
+						if (key_value) {
```

Comment:
Presumably this is rare that the error flags, and I'm guessing the processing in `_try_to_find_constant_value_for_expression()` is more than the old `if (keys.has(keyName))`?

If that is the case would it be an idea to do a quick reject with the old `if (keys.has(keyName))` before calling `_try_to_find_constant_value_for_expression()`?

Reply:
No, there is no need to do that. If you check the definition of  `_try_to_find_constant_value_for_expression()`, it first checks whether the current token is a constant and returns it's value. 

In fact, `_try_to_find_constant_value_for_expression()` is not a replacement of old `if (keys.has(keyName))`, it's meaning is to correctly evaluate the value of the token, so in fact it just improves that if-condition (it's still there, just few lines below).